File: src/api/configDetection.ts
/**
 * Typed Tauri command wrappers for the Config Detection Engine.
 *
 * Keep names aligned with `src-tauri/src/commands/config_detection.rs`.
 */

import { invoke } from "@tauri-apps/api/core";
import type { ConfigDetectionResult } from "../types/configDetection";

export async function detectConfigPlatform(
  configText: string,
): Promise<ConfigDetectionResult> {
  return invoke<ConfigDetectionResult>("detect_config_platform", {
    configText,
  });
}

File: src-tauri/src/commands/config_detection.rs
//! Tauri commands for the Config Detection Engine.

use crate::engines::config_detection::{self, ConfigDetectionResult};

#[tauri::command]
pub fn detect_config_platform(config_text: String) -> ConfigDetectionResult {
    config_detection::detect_config_platform(&config_text)
}
